testplan/common/utils/timing.py

```diff
@@ -92,10 +95,14 @@ def timeout(seconds, err_msg='Timeout after {} seconds.'):
     def timeout_decorator(func):
         """"""
         def _new_func(result, old_func, old_func_args, old_func_kwargs):
-            result.append(old_func(*old_func_args, **old_func_kwargs))
+            try:
+                result.append(old_func(*old_func_args, **old_func_kwargs))
+            except Exception as exp:
+                result[0] = False
+                result.append(format_trace(inspect.trace(), exp))
```

Comment:
It's not very important but I think that `traceback.format_exc()` would be simpler here, I don't see why we need a custom function for formatting stack traces when it's right there in the standard library.

Reply:
I' m not sure, I saw that the __exit__() function of Exception Logger calls this format_trace. Is it some different ?

Reply:
I don't think we need it, it just does the same as standard library function.

Reply:
I will check it and if not necessary I will remove it.

Reply:
Hi Ryan, as I check again, **format_trace(inspect.trace(), exc))** is used at many places in open source repo. To avoid making it not stable so I won't change it in this pull request. If you though it's necessary then another PR can be opened.

Reply:
that's fine, I wasn't suggesting you remove it in this PR. I was more just asking the broader question of do we need it, and do you need to use it right here. We can look at removing in another PR, if it can be replaced by the traceback module.